src/services/item/plugins/invitation/index.ts

```diff
@@ -192,6 +210,59 @@ const plugin: FastifyPluginAsync = async (fastify) => {
       reply.status(StatusCodes.NO_CONTENT);
     },
   );
+
+  fastify.post<{ Params: { itemId: string } }>(
+    '/:itemId/enroll',
+    {
+      schema: enroll,
+      preHandler: [isAuthenticated, matchOne(validatedMemberAccountRole)],
+    },
+    async ({ user, params }) => {
+      const member = notUndefined(user?.account);
+      assertIsMember(member);
+      const { itemId } = params;
+
+      return await db.transaction(async (manager) => {
+        const repositories = buildRepositories(manager);
+
+        const item = await itemService.get(
+          member,
+          repositories,
+          itemId,
+          PermissionLevel.Read,
+          false,
+        );
+
+        if (!(await itemLoginService.getByItemPath(repositories, item.path))) {
+          const error = new ItemLoginSchemaNotFound();
+          error.statusCode = StatusCodes.FORBIDDEN;
```

Comment:
I'm wondering if we shouldn't throw another error here? ErrorNotFound is a 404, but the status code is updated to 403?

Reply:
I would indeed throw sth like "cannot enroll" 403